File: src/main/java/com/mcworkshop/wehcm/web/rest/controller/PassiveFlowController.java
package com.mcworkshop.wehcm.web.rest.controller;

import com.mcworkshop.wehcm.core.domain.message.PassiveMessage;
import com.mcworkshop.wehcm.core.persistence.PassiveMessageRepository;
import com.mcworkshop.wehcm.core.service.PassiveMessageService;
import com.mcworkshop.wehcm.web.rest.resource.PassiveMessageResource;
import com.mcworkshop.wehcm.web.rest.resource.asm.PassiveMessageResourceAsm;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.PropertySource;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import javax.transaction.Transactional;
import java.util.UUID;

import static com.mcworkshop.wehcm.core.domain.message.Message.MESSAGE_KEY_MESSAGE_OID;

/**
 * Created by markfredchen on 6/27/15.
 */
@Controller
@PropertySource("classpath:application.properties")
public class PassiveFlowController {

    @Value("${wehcm.account.path}")
    private String accountPath;

    @Autowired
    PassiveMessageRepository messageRepository;

    @Autowired
    PassiveMessageService passiveMessageService;

    @Transactional
    @RequestMapping(value = "/passive/flow/{messageOID}", method = RequestMethod.GET)
    @ResponseBody
    public PassiveMessageResource loadPassiveMessage(@PathVariable("messageOID") UUID messageOID) {
        PassiveMessage passiveMessage = messageRepository.findOneByMessageOID(messageOID);
        return new PassiveMessageResourceAsm().toResource(passiveMessage);
    }


    @RequestMapping(value = "/passive/flow/action", method = RequestMethod.POST)
    @ResponseBody
    public String handlePassiveMessageAction(@RequestBody String data) {
        JSONObject dataJSON = new JSONObject(data);
        UUID messageOID = UUID.fromString(dataJSON.getString(MESSAGE_KEY_MESSAGE_OID));
        String action = dataJSON.getString(PassiveMessage.PASSIVE_MESSAGE_KEY_ACTION);
        passiveMessageService.handlePassiveMessageAction(messageOID, action);

        JSONObject result = new JSONObject();
        result.put("messageOID", data);
        result.put("status", "OK");
        return result.toString();
    }
}

File: src/main/webapp/app/wechat/flow/passive/passive.flow.controller.js
/**
 * Created by markfredchen on 6/27/15.
 */
'use strict';
angular.module('wehcmApp')
    .controller('PassiveFlowController', function ($scope, $http, $location, angularLoad, $state, data) {

        $scope.config = data;

        $scope.init = function () {
            console.log($scope.messageOID);
            $http.get('/passive/flow/' + $scope.messageOID)
                .success(function (data, status, headers, config) {
                    $scope.config = data;
                }).error(function (data, status) {
                    if (status == 405) {

                    }
                });
        };

        $scope.makeAction = function (action) {
            console.log($scope.config.messageOID);
            $http.post('/passive/flow/action', {messageOID: $scope.config.messageOID, action: action})
                .success(function (data, status) {
                    $state.go('passive-flow-success');
                })
        };
    });
